### PR TITLE
feat!: add ability to encrypt seed words

### PR DESCRIPTION
**Pull request by SWvheerden**
Description
---
allow the use of encrypted seed words.

### CODE REVIEW

File: applications/minotari_console_wallet/src/lib.rs [hunk 1]
```diff
@@ -284,7 +284,7 @@ fn get_recovery_seed(
 ) -> Result<Option<CipherSeed>, ExitError> {
     if matches!(boot_mode, WalletBoot::Recovery) && !matches!(wallet_type, Some(WalletType::Ledger(_))) {
         let seed = if let Some(ref seed_words) = cli.seed_words {
-            get_seed_from_seed_words(seed_words)?
+            get_seed_from_seed_words(seed_words, None)?
```
Comment (hansieodendaal) [inline applications/minotari_console_wallet/src/lib.rs:287]: I think the passphrase should be an option here as well, and be included in function `prompt_private_key_from_seed_words()`.

**Review by hansieodendaal — COMMENTED** [review]
Apart from fixing tests, one functional use comment, otherwise it seems fine.

File: applications/minotari_console_wallet/src/lib.rs (hunk 1 above)
Comment (SWvheerden) [reply applications/minotari_console_wallet/src/lib.rs:287]: This is fine I think, its an internal helper function, console wallet does not yes use the encrypted seed words

Comment (hansieodendaal) [reply applications/minotari_console_wallet/src/lib.rs:287]: But what if you want to convert a mobile wallet with encrypted seed words to a console wallet?

**Review by hansieodendaal — DISMISSED** [review]
utACK

File: applications/minotari_console_wallet/src/lib.rs (hunk 1 above)
Comment (SWvheerden) [reply applications/minotari_console_wallet/src/lib.rs:287]: https://github.com/tari-project/tari/issues/6572

**Pull request merged**